src/app-layout/visual-refresh-toolbar/contexts.ts

```diff
@@ -0,0 +1,12 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React from 'react';
+
+import { awsuiPluginsInternal } from '../../internal/plugins/api';
+
+interface BreadcrumbsSlotContextType {
+  isInToolbar: boolean;
+}
+
+export const BreadcrumbsSlotContext =
+  awsuiPluginsInternal.sharedReactContexts.createContext<BreadcrumbsSlotContextType>(React, 'BreadcrumbsSlotContext');
```

Comment:
This is needed, because by default `React.createContext` produces a new context for each call. In our case, there could be multiple copies of this file, which will not allow context providers connect to its consumers. 

We solve it by caching context instances by their name

Reply:
Yes, because `awsuiPluginsInternal` object is deduplicated. https://github.com/cloudscape-design/components/blob/01aecc9aefe0c976f3a586e8ba6917d98540c237/src/internal/plugins/api.ts#L62-L63